Cargo.toml

```
@@ -41,6 +41,13 @@ tokio = { version = "1.18.0", features = [ "full" ] }
 url = "2.2.2"
 xz2 = "0.1.6"
 zip = "0.6.2"
+# zstd is also depended by zip.
+# Since zip 0.6.2 depends on zstd 0.10.0, we also have to use 0.10.0 here,
+# otherwise there will be a link conflict.
+#
+# Enable feature bindgen to generate C bindings.
+# Enable feature zstdmt to enable multithreading in libzstd.
+zstd = { version = "0.10.0", features = [ "bindgen", "zstdmt" ], default-features = false }
```

```suggestion
zstd = { version = "0.10.0", features = [ "bindgen", "zstdmt", "pkg-config" ], default-features = false }
```

do note that this will still not work, you'll need something like I mentioned in https://github.com/cross-rs/cross/issues/731#issuecomment-1141882347

which is a custom Dockerfile to add the missing dependencies

---

Thanks!

However, I still don't think it is a good idea to have a runtime dependency here, so I will explore other options to fix this problem, like abandoning the glibc and goes for musl-only for these two targets.

---

it would not be a runtime dependency :)

---

mb! I meant zstd-sys

---

@Emilgardis I checked the `build.rs` and it [panic if `pkg-config` failed to probe for `zstd`](https://github.com/gyscos/zstd-rs/blob/56c8b52b6d448e376e746fc062077f68febc03a3/zstd-safe/zstd-sys/build.rs#L48).

---

yep, and to solve that, you need to have libzstd-dev installed in the builders environment, which is done with supplying a custom docker image to cross

---

It [skips building `zstd` if `pkg-config` is enabled](https://github.com/gyscos/zstd-rs/blob/56c8b52b6d448e376e746fc062077f68febc03a3/zstd-safe/zstd-sys/build.rs#L202).